modules/apps/29-fee/keeper/keeper.go

```diff
@@ -162,12 +160,9 @@ func (k Keeper) GetAllRelayerAddresses(ctx sdk.Context) []types.RegisteredRelaye
 
 	var registeredAddrArr []types.RegisteredRelayerAddress
 	for ; iterator.Valid(); iterator.Next() {
-		keySplit := strings.Split(string(iterator.Key()), "/")
-
-		addr := types.RegisteredRelayerAddress{
-			Address:             keySplit[1],
-			CounterpartyAddress: string(iterator.Value()),
-			ChannelId:           keySplit[2],
+		addr, err := types.ParseKeyCounterpartyRelayer(string(iterator.Key()), string(iterator.Value()))
```

Comment:
Let's call this `registeredArr`. I expect a variable like `addr` being a byte array or string not a struct

Reply:
i'll just have the function return the address and channel-id values from the key